stage1_fly/run/main.go

```diff
@@ -0,0 +1,381 @@
+		}
+	}
+
+	// Gather host mounts which we make MS_SHARED if passed as a volume source
+	hostMounts, err := getHostMounts()
+	if err != nil {
+		log.Fatalf("fly: error gathering host mounts")
+	}
+
+	argFlyMounts := []flyMount{}
+	var flags uintptr = syscall.MS_BIND // TODO: allow optional | syscall.MS_REC
+	for _, tuple := range namedVolumeMounts {
+		if _, isHostMount := hostMounts[tuple.V.Source]; isHostMount {
+			// Mark the host mount as SHARED so the container's changes to the mount are propagated to the host
+			argFlyMounts = append(argFlyMounts,
+				flyMount{"", "", tuple.V.Source, "none", syscall.MS_REC | syscall.MS_SHARED},
+			)
+		}
+		argFlyMounts = append(argFlyMounts,
+			flyMount{tuple.V.Source, rfs, tuple.M.Path, "none", flags},
+		)
+
+		if tuple.V.ReadOnly != nil && *tuple.V.ReadOnly {
+			argFlyMounts = append(argFlyMounts,
+				flyMount{"", rfs, tuple.M.Path, "none", flags | syscall.MS_REMOUNT | syscall.MS_RDONLY},
+			)
+		}
+	}
+	return argFlyMounts, nil
+}
+
+func stage1() int {
+	uuid, err := types.NewUUID(flag.Arg(0))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "UUID is missing or malformed")
+		return 1
+	}
+
+	root := "."
+	p, err := stage1commontypes.LoadPod(root, uuid)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to load pod: %v\n", err)
+		return 1
+	}
+
+	if len(p.Manifest.Apps) != 1 {
+		log.Fatalf("Flavor %q only supports 1 application per Pod for now.", flavor)
+	}
+
+	// TODO: insert environment from manifest
+	env := []string{"PATH=/bin:/sbin:/usr/bin:/usr/local/bin"}
+	args := p.Manifest.Apps[0].App.Exec
+	rfs := filepath.Join(common.AppPath(p.Root, p.Manifest.Apps[0].Name), "rootfs")
+
+	// set close-on-exec flag on RKT_LOCK_FD so it gets correctly closed when invoking
+	// network plugins
+	lfd, err := common.GetRktLockFD()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to get rkt lock fd: %v\n", err)
+		return 1
+	}
+
+	if err := sys.CloseOnExec(lfd, true); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to set FD_CLOEXEC on rkt lock: %v\n", err)
+		return 1
+	}
+
+	argFlyMounts, err := evaluateMounts(rfs, string(p.Manifest.Apps[0].Name), p)
+	if err != nil {
+		log.Fatalf("Error evaluating mounts: %v", err)
+	}
+
+	effectiveMounts := append(
+		[]flyMount{
+			{"", "", "/dev", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/dev", rfs, "/dev", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"", "", "/proc", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/proc", rfs, "/proc", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"", "", "/sys", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/sys", rfs, "/sys", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"tmpfs", rfs, "/tmp", "tmpfs", 0},
+		},
+		argFlyMounts...,
+	)
+
+	for _, mount := range effectiveMounts {
+		var (
+			err            error
+			hostPathInfo   os.FileInfo
+			targetPathInfo os.FileInfo
+		)
+		if mount.HostPath != "" && strings.HasPrefix(mount.HostPath, "/") {
```

Comment:
The `mount.HostPath != ""` check is unnecessary - if it is an empty string then `strings.HasPrefix(..., "/")` will catch it.